hangman.py

```diff
@@ -0,0 +1,200 @@
+      lowercase
+    letters_guessed: list (of letters), which letters have been guessed so far;
+      assumes that all letters are lowercase
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    pass
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    pass
+
+
+def get_available_letters(letters_guessed):
+    '''
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    pass
+
+
+def warnings(value_mistake, warnings_point, guesses_point):
+    """
+    Subtract warnings point and output name of error
+    """
+    name_mistake = f"Oops! {value_mistake}"
+    if warnings_point > 0:
+        warnings_point -= 1
+        print(name_mistake, "\nYou have", warnings_point, "warnings left:", end=" ")
+    else:
+        guesses_point -= 1
+        print("Oops! You have no warnings left, so you lose one guess:", end=' ')
+    return warnings_point, guesses_point
+
+
+def letter_in_word(secret_word, letter, list_vowels, letter_guessed, guesses_point, letters_guessed):
+    """
+    Check if letter is word, else subtracts point
+    """
+    letters_guessed.append(letter)
+    if letter in secret_word:
+        print("Good guess:", get_guessed_word(secret_word, letters_guessed))
+    else:
+        if letter in list_vowels:
+            guesses_point -= 2
+        else:
+            guesses_point -= 1
+        print("Oops! That letter is not in my word:", get_guessed_word(secret_word, letters_guessed))
+    return letters_guessed, guesses_point
+
+
+def hangman(secret_word, request, list_vowels, warnings_point, letters_guessed, guesses_point):
+    '''
+    Game Hangman. "request" show hangman(classic) or hangman with hints.
+    Print information for user, start loop, check user choose hangman or hangman with hints,
+    Check input letter and call letter_in_word, check if user guessed word.
+    '''
+    print("I am thinking of a word that is", len(secret_word), "letters long")
+    print("You have", warnings_point, "warnings left")
+    while guesses_point > 0:
+        print("-------")
+        print("You have", guesses_point, "guesses left")
+        print("Available letters:", end=" ")
+        get_available_letters(letters_guessed)
+        letter = input("Please guess a letter:")
+        if request is True and letter == "*":
```

Comment:
It is not recommended to compare with True or False. Even more not recommended to check it with `is` operator